cpp/src/gandiva/engine.h

```diff
@@ -23,9 +23,9 @@
 #include <string>
 #include <vector>
 
-#include "arrow/status.h"
 #include "arrow/util/macros.h"
 
+#include "gandiva/arrow.h"
```

Comment:
Hmm... can we keep the includes as minimal as possible? Arrow is already slow enough to build.

Reply:
clangd has this option to automatically add header when auto-completing. I disabled it.